libafl/src/mutators/token_mutations.rs

```diff
@@ -79,60 +82,49 @@ impl Tokens {
         Ok(self)
     }
 
-    /// Build tokens from autotokens
-    pub fn parse_autotokens(mut self, autotoken: TokenSection) -> Result<Self, Error> {
-        self.add_from_autotokens(autotoken)?;
-        Ok(self)
-    }
-
+    /// Create a token section from a start and an end pointer
     /// Reads from an autotokens section, returning the count of new entries read
-    pub fn add_from_autotokens(&mut self, autotoken: TokenSection) -> Result<usize, Error> {
-        if cfg!(target_os = "linux") {
-            let mut entries = 0;
-            let token_start = autotoken.start;
-            let token_stop = autotoken.stop;
-            let section_size: usize = unsafe { token_stop.offset_from(token_start) }
-                .try_into()
-                .unwrap();
-            // println!("size: {}", section_size);
-            let slice = unsafe { from_raw_parts(token_start, section_size) };
-
-            let mut head = 0;
-
-            // Now we know the beginning and the end of the token section.. let's parse them into tokens
-            loop {
-                if head >= section_size {
-                    // Sanity Check
-                    assert!(head == section_size);
-                    break;
-                }
-                let size = slice[head] as usize;
-                head += 1;
-                if size > 0 {
-                    self.add_token(&slice[head..head + size].to_vec());
-                    #[cfg(feature = "std")]
-                    println!(
-                        "Token size: {} content: {:x?}",
-                        size,
-                        &slice[head..head + size].to_vec()
-                    );
-                    head += size;
-                    entries += 1;
-                }
+    #[must_use]
+    #[cfg(linux)]
+    pub unsafe fn from_ptrs(token_start: *const u8, token_stop: *const u8) -> Result<Self, Error> {
+        let ret = Self::default();
+        if token_start == null() || token_stop == null() {
+            return Ok(ret)
```

Comment:
I feel return Err() is better here

Reply:
Because the build failed earlier?

Reply:
See efe05d0b0cc91e04e2a3952aa0ff71ee96d3a663

Reply:
no...
I just thought if `token_stop <= token_start ` results in an error. then nullptr should do so as well.